### ISSUE TITLE
Nodejs botbuilder promises don't Send messages

### ISSUE THREAD

**Issue by AndyCross** [Microsoft/BotBuilder#1068]
I'm looking for advice on handling promise resolutions. Here's a really basic sample that will output "start of test" but not "end of test". 

```js
var builder = require('botbuilder');
var connector = new builder.ConsoleConnector().listen();
var bot = new builder.UniversalBot(connector);

bot.dialog('/', new builder.IntentDialog()
    .matches(/test/i, function (session) {
    session.send("start of test");
    var p = new Promise(function (resolve, reject) {
        setTimeout(resolve, 5000, 15);
    });
    p.then(function () {
        session.send("end of test");
        console.dir(session);
    });
})
    .onDefault(function (session) {
    session.send("I didn't get that; try /test");
}));
```

The session object is not null and seems in a good state, but no sending occurs when we reach the end of test position. 

Any thoughts welcome.

**Comment by Thegaram** [comment]
This is a bug. In some cases, `ConsoleConnector` leaves the `Session` object in an inconsistent state, thus preventing subsequent messages from being sent. This should be fixed in the next release, #1023 is the related issue. (Your code should work as expected with `ChatConnector`.)

**Comment by AndyCross** [comment]
Thanks for the insight @Thegaram 

I'll await the fix in the release.

**Comment by palmerabollo** [comment]
@AndyCross It is fixed in botbuilder 3.3.0.
